includes/Generator/Generator.php

```diff
@@ -11,13 +11,25 @@
  * Data generator base class.
  */
 abstract class Generator {
-
+	/**
+	 * Maximum number of objects that can be generated in one batch.
+	 */
 	const MAX_BATCH_SIZE = 100;
 
+	/**
+	 * Dimension, in pixels, of generated images.
+	 */
 	const IMAGE_SIZE = 700;
 
 	/**
-	 * Holds the faker factory object.
+	 * Are we ready to generate objects?
+	 *
+	 * @var bool
+	 */
+	protected static $ready = false;
```

Comment:
All generators would "share" this `$ready` property (given its static nature). This doesn't seem to have any serious impact, but I wonder if this is truly what you intended here vs each generator having its own instance.

Just want to double check as the `protected` seems to indicate the idea was for each generator to have its own initialization state, though I might be wrong :)

Reply:
🤔 It's a good point. Right now, I think it's actually totally fine if the property (and the `$faker` static property) are shared between generators, because they can all use the same instance, and the email hooks only need to be removed once. However, I think I probably did intend for each generator to have its own instance, and other people working with this code might make the same assumption.

Could this be fixed by changing to `static::$ready`? As in

```php
	protected static function initialize() {
		if ( true !== static::$ready ) {
			// stuff
		}

		static::$ready = true;
	}
```

Reply:
Clarified generator initialization in 7b93a87